Tests/UnitTests/Plugins/Detray/DetrayGeometryConverterTests.cpp

```diff
@@ -0,0 +1,302 @@
+  Transform3 m_transform;
+  CylinderVolumeBounds m_volumeBounds;
+  surface_bounds_type m_surfaceBounds;
+  std::string m_name;
+};
+
+BOOST_AUTO_TEST_SUITE(DetrayConversion)
+
+BOOST_AUTO_TEST_CASE(DetrayTransformConversion) {
+  auto transform = Transform3::Identity();
+  transform.pretranslate(Vector3(1., 2., 3.));
+  transform.rotate(Eigen::AngleAxisd(M_PI / 2., Vector3::UnitZ()));
+
+  detray::io::transform_payload payload =
+      DetrayGeometryConverter::convertTransform(transform);
+  // Transform is correctly translated
+  CHECK_CLOSE_ABS(payload.tr[0u], 1.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.tr[1u], 2.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.tr[2u], 3.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  // Rotation is correctly translated
+  const auto rotation = transform.rotation();
+  CHECK_CLOSE_ABS(payload.rot[0u], rotation(0, 0),
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.rot[1u], rotation(0, 1),
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.rot[2u], rotation(0, 2),
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.rot[3u], rotation(1, 0),
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.rot[4u], rotation(1, 1),
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.rot[5u], rotation(1, 2),
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.rot[6u], rotation(2, 0),
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.rot[7u], rotation(2, 1),
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.rot[8u], rotation(2, 2),
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+}
+
+BOOST_AUTO_TEST_CASE(DetrayMaskConversion) {
+  // Placeholder, masks are tested for the moment through the DetrayJsonHelper,
+  // this code will move over here when the "toJsonDetray" will become
+  // deprecated
+}
+
+BOOST_AUTO_TEST_CASE(DetraySurfaceConversion) {
+  // Translate a cylinder
+  auto cylinderSurface = Acts::Surface::makeShared<CylinderSurface>(
+      Transform3::Identity(), std::make_shared<CylinderBounds>(20., 100.));
+
+  auto sgID = Acts::GeometryIdentifier().setSensitive(1);
+  cylinderSurface->assignGeometryId(sgID);
+
+  detray::io::surface_payload payload = DetrayGeometryConverter::convertSurface(
+      tContext, *cylinderSurface, false);
+
+  // Check the payload
+  BOOST_CHECK(!payload.index_in_coll.has_value());
+  BOOST_CHECK(payload.mask.shape == detray::io::shape_id::cylinder2);
+  BOOST_CHECK_EQUAL(payload.source, sgID.value());
+  BOOST_CHECK(payload.type != detray::surface_id::e_portal);
```

Comment:
If the cylinder can be set to sensitive now, then we can make the check a bit tighter:
```suggestion
  BOOST_CHECK(payload.type == detray::surface_id::e_sensitive);
```